### PR TITLE
[MODULAR] Complete DS2 remap + 2 new syndicate spawners

### PR DESCRIPTION
**Pull request by zydras**
## About The Pull Request

Congratulations, Operatives. DS2 has been selected for an upgrade and has now been turned into a fully operational spaceship. 
Syndicate Command has found fitting to hire more personnel to your cryobanks, and you will now see Mining Officers and Corporate Liaisons staff your station. 

<details>
 <summary>The full map</summary>

image
</details>

<details>
 <summary>New SecHUD icons & IDs</summary>
 
Screenshot (1650)
Screenshot (1649)
</details>

## How This Contributes To The Skyrat Roleplay Experience

* DS2 had issues with being sprawled out and without any proper objective or game loop compared to Interdyne. This led to DS2 oftentimes being deserted in favor of everyone being on Interdyne. A complete remap ought to fix that.
* The addition of 2 new spawner types (Mining Officer & Corp. Liaison) will further increase the ability DS2 has to roleplay. Mining Officer is simply the Syndicate version of a Shaft Miner while Corporate Liaison is akin to a Syndicate Head of Personnel, now with a fitting quarters.

## Proof of Testing

<details>
 <summary>They said 'we need proof of testing', they're gonna get proof of testing.</summary>

Screenshot (1614)
Screenshot (1615)
Screenshot (1617)
Screenshot (1618)
Screenshot (1619)
Screenshot (1620)
Screenshot (1621)
Screenshot (1631)
Screenshot (1623)
Screenshot (1624)
Screenshot (1625)
Screenshot (1626)
Screenshot (1627)
Screenshot (1629)
Screenshot (1630)
</details>

## Changelog

:cl:
add: DS-2 has been fully retrofitted into a working spaceship and have had their ID trims fully updated.
add: DS-2 is now staffed with two Mining Officers and a Corporate Liaison.
add: Donk Co. has provided DS-2 with all new outfits for most of its members.
balance: Many of the DS-2 gear the ship starts with has been either cut, modified or rebalanced. No more loot pinata. 
del: DS-2 has had its pool decommissioned and was told to "visit the space hotel". Rude!
/:cl:

### CODE REVIEW

**Comment by Zonespace27** [comment]
the ds-2 mapping curse has finally been broken

File: modular_skyrat/master_files/code/datums/id_trim/syndicate.dm
Comment (RimiNosha) [inline modular_skyrat/master_files/code/datums/id_trim/syndicate.dm]: Heads up, it might be an idea to append DS2 to the DS2 roles so they aren't confused.

Comment (zydras) [reply modular_skyrat/master_files/code/datums/id_trim/syndicate.dm]: I believe none of the DS2 jobs actually have a DS2 suffix. Besides, they're on the black ID cards, figures it's easy to figure out IC.
Unless you meant codewise, like '/datum/id_trim/syndicom/skyrat/assault/ds2miner' ?

Comment (zydras) [reply modular_skyrat/master_files/code/datums/id_trim/syndicate.dm]: Going to turn the 'skyrat/assault' part into 'skyrat/ds2'

**Comment by unit0016** [comment]
Unfathomably based. Godspeed.

**Comment by Ebin-Halcyon** [comment]
Awww yiss.

**Comment by Tastyfish** [comment]
Based testing screenshots

**Comment by zydras** [comment]
<details>
 <summary>I Have Become Victim Of Society's Pressure (2 discord members) And Added Two Brig Cells (good ending) (see image)</summary>

Screenshot (1637)
</details>

**Comment by YakumoChen** [comment]
What do the miners do? Most space mining is limited to ruins, the most important of which are valuable to preserve for specific ghost roles (like Tarkon). Otherwise minerals are only available through looting.

Interdyne has its own miners, which make sense for lavaland, but ore silos cannot take or receive from other Z levels. For the same reason, having both an Interdyne and DS2 research is fairly redundant.

Iffy on requiring a ghost role to do power gen setup as well, Jump-starting power can be an extra pain if players only start spawning mid-round.

**Comment by YakumoChen** [comment]
Regardless of the comment above add `closes #17197 ` to your PR body please since my PR will be irrelevant if this is merged

**Comment by zydras** [comment]
> What do the miners do? Most space mining is limited to ruins, the most important of which are valuable to preserve for specific ghost roles (like Tarkon). Otherwise minerals are only available through looting.

I don't really intend miners to go and do space mining, and for right now I'm mostly intending on them using the quantum pad to interdyne to acquire lavaland ores.

> Interdyne has its own miners, which make sense for lavaland, but ore silos cannot take or receive from other Z levels. For the same reason, having both an Interdyne and DS2 research is fairly redundant.

Interdyne will - in the near future - be separated from DS2 as it becomes a neutral entity. This means that, effectively, DS2 would have zero means to get new materials. Another issue is that DS2, if it has no miners, has to siphon Interdyne of their materials which isn't exactly bueno either. On that note, a PR in the future might rise up to rebalance the relationship between both ghostroles where Interdyne is Medical-focused (viro, xenobio) & DS2 is Engineering/Science-focused (R&D, Engineering).

> Iffy on requiring a ghost role to do power gen setup as well, Jump-starting power can be an extra pain if players only start spawning mid-round.

This _is_ true, and as such I'll have to think about it deeper, but in general I'm mostly trying to give DS2 content to do that doesn't involve "Take the quickest shuttle to Interdyne", and some actual content that isn't immediately given to you could be appreciated. I'll think about maybe readding the bluespace (magic) SMES that generates power, but I do need to see if said free power generation is tweakable.

> Regardless of the comment above add closes #17197 to your PR body please since my PR will be irrelevant if this is merged

....I'm not familiar enough with github to figure out how that works, unless you intend on closing your PR yourself? But yes, I'll add that.

**Comment by zydras** [comment]
On-topic:
<details>
 <summary>The syndicate has massive drip</summary>
 
image
</details>

**Comment by YakumoChen** [comment]
> ....I'm not familiar enough with github to figure out how that works, unless you intend on closing your PR yourself? But yes, I'll add that.

All you have to do is add the text and github does the rest (what you did is fine, you can tell its working by hovering the word closes), it just makes my PR automatically close when this is merged

**Comment by zydras** [comment]
**Please do NOT merge before thursday. I'll be away from my workstation.**
(Putting my todo list here so I don't forget):

- [x] Update ALL jobs to have the right trims (thank you Gandalf <3)
- [x] Give Scientists the right labcoat in loadout
- [x] Remove the mask Mining Officers get (and maybe? put it in their locker)
- [x] Giving miners a renamed cargo armband, for aesthetic purposes
- [x] Give Service crew a white cap & chef apron in loadout
- [x] Fixing a typo in ghostrole flavor text (sector 13 instead of Sector 13)
- [x] Add papers that give instructions to the Admiral, Corp. Liaison and the Master At Arms
- [x] Remove the Combat gloves plus+ gloves from the armory
- [x] Add a pair of Combat gloves plus+ and an APS to the Admiral's locker (and remove the APS from the admiral's loadout)
- [x] Give the MAA the krav maga implant in their loadout

**Comment by axietheaxolotl** [comment]
i like dis a lot

**Comment by zydras** [comment]
(moved content of this comment into main body)

**Comment by unit0016** [comment]
> SecHUD icons & IDs

FYI, in #14965 the way ID Cards handle colors was updated - doesn't look like the modular icons are compliant with it

**Comment by zydras** [comment]
> doesn't look like the modular icons are compliant with it

Ope, good catch, let me check that PR

**Comment by zydras** [comment]
Another todo:

- [x] Check if engis can open the welding & electrical supplies lockers (access, mostly)
- [x] Put Granite's special Syndicate Cyborg Spawner into robotics
- [x] Remove/Replace the Security Checkpoint locker with something else (redundant)(still questioning)
- [x] Fix missing text on the crumpled paper in the Vault
- [x] Fix an overlapping air alarm & pepperspray dispenser in the brig
- [x] Remove a misplaced disabler in the armory

**Comment by zydras** [comment]
@unit0016 I've done the checks and I believe the PR you linked only affects the nonmodular IDs & regular stationside stuff. SR has a variety of custom ID trims & cards (in card.dmi and trim.dmi) that don't rely on being var-editable like that. Besides, i'm not a good enough coder to understand the whole gist of it, lmao,

**Comment by unit0016** [comment]
> @unit0016 I've done the checks and I believe the PR you linked only affects the nonmodular IDs & regular stationside stuff. SR has a variety of custom ID trims & cards (in card.dmi and trim.dmi) that don't rely on being var-editable like that. 

Yeah, thats the problem. Y'all have been missing out on spray can recolorability and didn't even notice, lmao. I guess I'll redownload SR's repo and take a crack at it once this is merged.

**Review by Paxilmaniac — CHANGES_REQUESTED** [review]
image
Put turrets on these two bits at the back, because it seems very easy to just avoid the turrets on the back of the ship otherwise.

image
What's with the random disabler on the floor?

image
While I can excuse most of the two thick walls on this, this is the only place where I won't. Seems awkward and players definitely will notice the strangeness here, try and find a way to remedy that?

image
Have you checked to make sure the nukie figures on top of this container layer on top of the container on mapload?

image
I know the old one had it but I really don't like having... I think this is just a varedited fluff structure? Real or not I also just don't think it should be here since it'll never be used, not once, and just exists to fill space in the vault.

image
Ditto on the fake (or real) nuke, I don't think we really need to have a fake nuke disk either.

image
Store uranium and plasma in a crate rather than open on a table, as its otherwise a radiation/fire hazard. A single random spark from a lightbulb and this room is a radioactive inferno.

image
image
The decals and tiles on the insides of these airlocks feel strange compared to the rest of the ship, I can't quite put my finger on why but I think you could try something similar to what I've done for airlocks before:
image
image

**Comment by zydras** [comment]
Excellent changes! I'll get to them once i'm home. The disabler was me fat-fingering and putting it there by sheer accident. I do believe the figurines layer properly, but i'll give it a second check. Good call on the missed turrets, and I'll probably remove the fake nuke silo, yes. Good call on the crate for the mats, too.

**Comment by zydras** [comment]
Third TODO list:
- [x] Remove the Cyborg code while I try to make this PR testmerge-ready
- [x] Make sure the Admiral gets his APS (or an equivalent weapon)
- [X] Add extra turrets to the back
- [x] Make sure sprite layering is correct on the Donk co. container
- [x] Fix the thick wall in dormitories
- [x] Fix the airlock decals
- [x] Add a crate to engineering to contain the immense aura of uranium
- [x] Drop the nuke silo & disk

**Comment by zydras** [comment]
<details>
 <summary>The Donk Co. container layers correctly, as assumed</summary>

Screenshot (1653)
</details>

**Comment by zydras** [comment]
Right. I planned to have Cyborgs for DS2, but I want to get this mapping put into the game soon enough for a testmerge. Cyborgs will be in a future PR, it is.

(By heavens have the linters work)

**Comment by zydras** [comment]
> makes commit
> closes computer
> immediately sees a row of firelocks that are wrong

Dismay. Pure dismay.
- [x] Fix the firelocks to the left of the bridge

**Review by Jolly-66 — DISMISSED** [review]
I looked over a few things.

<details>
<summary> Atmos </summary

The mixers are not accessible.
image

</details>

<details>
<summary> Wall stuff </summary>

You want to avoid clogging walls up as such. Ghosts mostly will see it, yes, but mapping it gets annoying needing to sift through wall mounted stuff. Generally, overlapping is a bad practice to do and should be avoided.
image

image

</details>

<details>
<summary> Other minor nitpicks </summary

The curtains should be mechanically controlled. This is just a semi-customary thing we do.
image

You don't need lattice under the engines. I'm pretty sure its whats causing the walls to connect.
image

Not mapping related by seeing these resprited would be nice. This would be @Ebin-Halcyon decision, however.
image

image


I have two qualms with this.
One, how the collectors are arranged. It creates an annoying small pathline to the right.
Two, the relay. Please. God. If this area is supposed to be lowly trafficked, fine, but its still an ear sore.
image

Near space on the thrusters please.
image

Disposal outlet should be a chute. Typically what it is.
image

Window isn't needed behind the disposal unit (ignore the fact I deleted it)
image

Don't know how I feel about this one bit of tile decals.
image

Could probably hookup a water tank to these. Just some food for thought.
image

Please use directional subtypes.
image

image

I also find it INSANELY funny that with xray cameras, you can effectively spy on people in the bathroom and dorm areas.

</details>

**Comment by Jolly-66** [comment]
Please keep in mind my review is based on commit [43286d0](https://github.com/Skyrat-SS13/Skyrat-tg/pull/17229/commits/43286d0d9d244c44da1bbe1fc9119f5d7d775b0d)

**Comment by zydras** [comment]
New (remaining) todo to sate the derg's requests:
- [x] Fix the cameras not being directional subtypes
- [x] Set engines as Nearstation
- [x] Hook up the medbay's shower to a water tank
- [x] Fix floor decals
- [x] Add a power monitor inside engineering
- [x] Relay - Be - Gone
- [x] Move wall machines to not layer one over the other
- [x] Check the mechanical curtains
- [ ] Wonder what I am doing with my life

**Comment by unit0016** [comment]
> You don't need lattice under the engines. I'm pretty sure its whats causing the walls to connect.

Engines just naturally smooth with shuttle walls for... whatever reason. They look best with a buffer in the form of a directional window in my experience.

**Comment by zydras** [comment]
Alright, this fella better pass all the checks because I'm positive it's ready for a test-merge. May or may not be away for the day (and maybe week, lmao,) So it's up to the keyholders to decide when to TM.

**Comment by zydras** [comment]
New spotted problems:
- [x] Atmos pipes are fucked up, fixed by adding a different color for N2
- [x] Missing gas analyzers in atmos
- [x] Misplaced tile types in the hallways meant for under the firelocks
- [x] Add pumps to the O2 and Scrubber intakes
- [x] Fix the Electrical Supplies locker's sprite missing a door
- [x] Fix DS2 turret controls being unreachable

**Comment by Youtubeboy139** [comment]
After test-playing this new map I noticed some things that need changing to increase playability.

First, move all these machines to a central location in this room, the Mech Fab output is blocked in it's current position and makes it impossible to fabricate anything. South side needs to be open in order for the Fabricator to work.
Sprite-0001

Second, replace the current SMES (highlighted in purple) with a Bluespace SMES (see the Lavaland Interdyne map to see what I mean) because currently the SMES dies about a third of the way though the shift and leaves DS2 without power.
Sprite-0002

Third, kinda weird to only have one miner supply closet and two miner spawns so I suggest moving the miner vendor to where the current supply closet it and moving the supply closet where the vendor currently is as illustrated below. Then move the tank storage to where the tank holder in the bottom left-hand corner of the cargo bay is, replace the random tank holder entirely and in the tank storage's place put a second miner supply closet.
Sprite-0003

Also, the new roles are spawning with twice the starting items (two internals boxes and two crowbars)

There might be more, I need to do more play testing

**Comment by Youtubeboy139** [comment]
Also suggest adding airlock markers here - similar to how the Brig on station works - to prevent prisoners from slipping out of perma
Sprite-0004

**Comment by zydras** [comment]
You make excellent points, @Youtubeboy139! Sadly i'll be away from home until wednesday PM EST. A few things,

- For the Mech fab, you can rotate the output direction by opening the panel (screwdriver) after it's fully constructed & then alt-click to change output direction. But, you _are_ right, I can definitely rearrange the space.
- I do know the SMES used to be a Bluespace variant, but the issue is that it entirely rules out the need for *more* power, which can be one of the major gameplay loops of a game (else, if everything is already done mechanics wise, there's little to do between roleplay interactions, or if you're alone). I could see if there's a specific var that dictates how much it charges itself, but I could use help figuring that out.
- Excellent suggestions for the mining tools & spawners, I'll be doing that.
- Also an excellent suggestion for the brig airlock, I completely forgot the directional helpers.

As for the double inventory spawning, this is a known bug and issue with ghostroles right now where everything in bags duplicate. The mob_spawn.dm templates for the ghostrole spawners strictly state 1 crowbar and 1 internals box, but something duplicates it. May be wise to do an issue report, if that's not already there.

Thank you so much for the feedback and playtesting <3

**Comment by Iajret** [comment]
Haven't had this much problems with power. There's additional RTG parts already and you can upgrade exiting ones too. But power is sure more scarce now. You can't autogrow every hydroponic's tray and build insane plumbing factories, but this is more of a good thing than bad tbh. Maybe you should just variable change RTGs a bit or smth. 200 kWts should be more than enough for someone who doesn't want to mess with solars and/or turbine. Or maybe even 150 kWts.

Speaking of turbine, it may be a good thing to generate power with. Same as on tarkon. But without plasma, only viable is to pump just oxygen from atmospherics room and make power this way... and it just feels wrong.

**Comment by zydras** [comment]
Okay, let's see, todo for this evening:
- [x] Add a turbine to engineering, modify south airlock
- [x] Modify the mining spawner area
- [x] Give the admiral an extra APS magazine in their locker
- [x] Modify atmos to have a plasma miner section, remove the return inlets (the miners do that job)
- [x] Add airlock cyclers to the permabrig
- [x] Reorganize robotics around to not break the exofab
- [x] Add shears, some fertilizer & cultivators to prison botany
- [x] Give the MAA the right camera console

**Comment by Zergspower** [comment]
> Okay, let's see, todo for this evening:


add in to give the MAA the right cam console, currently it can only see Interdyne

**Comment by zydras** [comment]
> add in to give the MAA the right cam console, currently it can only see Interdyne

I missed that, thanks!!

**Comment by zydras** [comment]
merge conflicts my beloathed

**Comment by zydras** [comment]
So let's say, hypothetically of course,

**Comment by zydras** [comment]
Yet another TODO:

- [x] Port Syndifox from paradise, resprite it and add it
- [x] Give the medical bay the syndicate access mappers (I forgor)
- [x] Move the radsuit closet from the atmos section to just besides the Suit Storage Module
- [x] Restructure the dorms to have a bay window
- [x] Remove a reference I don't actually want there
- [x] Make the DS2 logo plaques an actual thing

**Comment by zydras** [comment]
Screenshot (1681)
(no you cannot roundstart control the syndifox)

**Comment by zydras** [comment]
another todo:
- [x] Give the prison drawing supplies
- [x] Add painting frames to DS2
- [x] Fishing portal (?)
- [x] Tobacco seeds in prison
- [x] Give the MAA a window
- [x] Add a Tegu
- [x] Make Syndifox have the syndicate faction marker (To not be shot by turrets)
- [x] Make Syndifox _not_ bleed blood
- [x] Remove a misplaced dog bed

**Comment by zydras** [comment]
Minor amounts of violence have been acquired.
image
image
These are identical statwise to regular riot shotguns, simply a reflavor.

**Comment by zydras** [comment]
oh right, i wanted to give the CL a lighter implant, I'll do that some day

**Comment by zydras** [comment]
<details>
 <summary>This mf got areas</summary>


</details>

File: modular_skyrat/modules/aesthetics/guns/code/guns.dm
Comment (Zonespace27) [inline modular_skyrat/modules/aesthetics/guns/code/guns.dm:73]: why not allow suppression?

File: modular_skyrat/modules/aesthetics/guns/code/guns.dm
Comment (Zonespace27) [inline modular_skyrat/modules/aesthetics/guns/code/guns.dm:72]: ditto

File: modular_skyrat/modules/mapping/code/mob_spawns.dm
Comment (Zonespace27) [inline modular_skyrat/modules/mapping/code/mob_spawns.dm]:
```suggestion
		/obj/item/storage/box/survival = 1,
```
do this for all of them

File: modular_skyrat/modules/mapping/code/mob_spawns.dm
Comment (Zonespace27) [inline modular_skyrat/modules/mapping/code/mob_spawns.dm]:
```suggestion
		/obj/item/crowbar=1
	)
```
same here

File: modular_skyrat/modules/mapping/code/mob_spawns.dm
Comment (Zonespace27) [inline modular_skyrat/modules/mapping/code/mob_spawns.dm]:
```suggestion
	id_trim = /datum/id_trim/syndicom/skyrat/ds2
```

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

File: modular_skyrat/modules/mapping/code/mob_spawns.dm
Comment (zydras) [reply modular_skyrat/modules/mapping/code/mob_spawns.dm]: Done,

File: modular_skyrat/modules/mapping/code/mob_spawns.dm
Comment (zydras) [reply modular_skyrat/modules/mapping/code/mob_spawns.dm]: Done, and,

File: modular_skyrat/modules/mapping/code/mob_spawns.dm
Comment (zydras) [reply modular_skyrat/modules/mapping/code/mob_spawns.dm]: Done!

**Comment by zydras** [comment]
todo:
- [ ] fix the CL being targetted by turrets (unsure what's the issue)
- [ ] fix the shotgun not being silence-able

File: modular_skyrat/modules/mapping/code/mob_spawns.dm
Comment (Zonespace27) [inline modular_skyrat/modules/mapping/code/mob_spawns.dm]:
```suggestion
		/obj/item/storage/box/survival = 1,
```
do for all

Comment (zydras) [reply modular_skyrat/modules/mapping/code/mob_spawns.dm]: oop, forgor, one moment.

Comment (zydras) [reply modular_skyrat/modules/mapping/code/mob_spawns.dm]: Done!

**Comment by Tastyfish** [comment]
On the live server, I spawned as a general DS-2 staff (kitchen spawner if it matters) and received 2 survival boxes and crowbars, but one of each of the other backpack contents. I don't know if this is the older duplicate backpack contents bug cropping up again, or specific to some or all of the DS-2 job/outfits, since I haven't tested it further.

**Comment by zydras** [comment]
> On the live server, I spawned as a general DS-2 staff (kitchen spawner if it matters) and received 2 survival boxes and crowbars, but one of each of the other backpack contents. I don't know if this is the older duplicate backpack contents bug cropping up again, or specific to some or all of the DS-2 job/outfits, since I haven't tested it further.

This is a known issue that affects all ghostroles with backpack contents, last I was told. I'm not sure what causes it, and I'm not talented enough to fix such bug by myself.

**Comment by Tastyfish** [comment]
> > On the live server, I spawned as a general DS-2 staff (kitchen spawner if it matters) and received 2 survival boxes and crowbars, but one of each of the other backpack contents. I don't know if this is the older duplicate backpack contents bug cropping up again, or specific to some or all of the DS-2 job/outfits, since I haven't tested it further.
> 
> This is a known issue that affects all ghostroles with backpack contents, last I was told. I'm not sure what causes it, and I'm not talented enough to fix such bug by myself.

fair

**Review by Paxilmaniac — APPROVED** [review]
Completed my requested map changes

**Comment by zydras** [comment]
Again, to-do list:
- [x] Change the hangar turfs to not be var-edited
- [x] Fix the engineering power monitor to be connected to the right power grid
- [x] Fix a typo on one of the medals
- [x] Actually fix the incinerator holofan transferring heat
- [x] Fix the air gasmixer
- [x] Fix a duplicate disposals outlet
- [x] Re-add the janicart

**Review by Zonespace27 — APPROVED** [review]
Awaiting map approval

**Review by Jolly-66 — APPROVED** [review]
Map checks out, doesn't have any major glaring issues.

Anything minor (aesthetically speaking) can be tweaked at a later date.

This has been TM'd for awhile now, and a lot of love and effort as gone into it.

Thank you for your work, Zydras.

**Comment by Zonespace27** [comment]
Merge time baybeee

**Comment by Zonespace27** [comment]
I did this on mobile which auto-deletes branches



If I really just deleted your 'master', whoops

**Comment by animalsandsuch** [comment]
love you babe

**Pull request merged**
